In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/factors-affecting-campus-placement/Placement_Data_Full_Class.csv


# Points to take into consideration
## Questions:
- Which factor influenced a candidate in getting placed?
- Does percentage matters for one to get placed?
- Which degree specialization is much demanded by corporate?
- Play with the data conducting all statistical tests.

## Regression tasks:
- Develop a simple linear regression model by estimating the model parameters.
- Show the regression plot
- Comment on the R-squared value of the model
- Test for significance of the regression co-efficient based on the summary

## Imports

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as py
import seaborn as sns
import math
import sklearn.model_selection as model_selection
import sklearn.preprocessing as preprocessing

from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score



## 1. Importing the data

In [3]:
data_df = pd.read_csv(r'/kaggle/input/factors-affecting-campus-placement/Placement_Data_Full_Class.csv')
data_df.head(20)

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0       1      M  67.00   Others  91.00   Others  Commerce     58.00   
1       2      M  79.33  Central  78.33   Others   Science     77.48   
2       3      M  65.00  Central  68.00  Central      Arts     64.00   
3       4      M  56.00  Central  52.00  Central   Science     52.00   
4       5      M  85.80  Central  73.60  Central  Commerce     73.30   
5       6      M  55.00   Others  49.80   Others   Science     67.25   
6       7      F  46.00   Others  49.20   Others  Commerce     79.00   
7       8      M  82.00  Central  64.00  Central   Science     66.00   
8       9      M  73.00  Central  79.00  Central  Commerce     72.00   
9      10      M  58.00  Central  70.00  Central  Commerce     61.00   
10     11      M  58.00  Central  61.00  Central  Commerce     60.00   
11     12      M  69.60  Central  68.40  Central  Commerce     78.30   
12     13      F  47.00  Central  55.00   Others   Science     65.00   
13     14      F  77.00  Central  87.00  Central  Commerce     59.00   
14     15      M  62.00  Central  47.00  Central  Commerce     50.00   
15     16      F  65.00  Central  75.00  Central  Commerce     69.00   
16     17      M  63.00  Central  66.20  Central  Commerce     65.60   
17     18      F  55.00  Central  67.00  Central  Commerce     64.00   
18     19      F  63.00  Central  66.00  Central  Commerce     64.00   
19     20      M  60.00   Others  67.00   Others      Arts     70.00   

     degree_t workex  etest_p specialisation  mba_p      status    salary  
0    Sci&Tech     No    55.00         Mkt&HR  58.80      Placed  270000.0  
1    Sci&Tech    Yes    86.50        Mkt&Fin  66.28      Placed  200000.0  
2   Comm&Mgmt     No    75.00        Mkt&Fin  57.80      Placed  250000.0  
3    Sci&Tech     No    66.00         Mkt&HR  59.43  Not Placed       NaN  
4   Comm&Mgmt     No    96.80        Mkt&Fin  55.50      Placed  425000.0  
5    Sci&Tech    Yes    55.00        Mkt&Fin  51.58  Not Placed       NaN  
6   Comm&Mgmt     No    74.28        Mkt&Fin  53.29  Not Placed       NaN  
7    Sci&Tech    Yes    67.00        Mkt&Fin  62.14      Placed  252000.0  
8   Comm&Mgmt     No    91.34        Mkt&Fin  61.29      Placed  231000.0  
9   Comm&Mgmt     No    54.00        Mkt&Fin  52.21  Not Placed       NaN  
10  Comm&Mgmt    Yes    62.00         Mkt&HR  60.85      Placed  260000.0  
11  Comm&Mgmt    Yes    60.00        Mkt&Fin  63.70      Placed  250000.0  
12  Comm&Mgmt     No    62.00         Mkt&HR  65.04  Not Placed       NaN  
13  Comm&Mgmt     No    68.00        Mkt&Fin  68.63      Placed  218000.0  
14  Comm&Mgmt     No    76.00         Mkt&HR  54.96  Not Placed       NaN  
15  Comm&Mgmt    Yes    72.00        Mkt&Fin  64.66      Placed  200000.0  
16  Comm&Mgmt    Yes    60.00        Mkt&Fin  62.54      Placed  300000.0  
17  Comm&Mgmt     No    60.00        Mkt&Fin  67.28  Not Placed       NaN  
18  Comm&Mgmt     No    68.00         Mkt&HR  64.08  Not Placed       NaN  
19  Comm&Mgmt    Yes    50.48        Mkt&Fin  77.89      Placed  236000.0

In [4]:
print(f'Shape:',data_df.shape)

Shape: (215, 15)


215 rows and 15 columns

In [5]:
# General overview of the dataset
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


We have 148 non-null columns for salary, so there are some missing values

**To get the exact number of missing values from the dataset:**

In [6]:
data_df.isnull().sum()

sl_no              0
gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

So we have 67 null values from the salary column

Whenever the status is **Not Placed**, we have a null salary value.

In [7]:
data_df['status'].value_counts()

Placed        148
Not Placed     67
Name: status, dtype: int64

Okay, so do all the values with the status **Not Placed** have the value of **NaN** as salary?

In [8]:
status_df = data_df[data_df['status']=='Not Placed']
status_df

print('shape of the status frame: ', status_df.shape)
print('Number of rows as status **Not Placed**: ',status_df['status'].count() )
print('Number of rows with salary as zero: ', status_df['salary'].isnull().count())

shape of the status frame:  (67, 15)
Number of rows as status **Not Placed**:  67
Number of rows with salary as zero:  67


In [9]:
status_df['salary']

3     NaN
5     NaN
6     NaN
9     NaN
12    NaN
       ..
198   NaN
201   NaN
206   NaN
208   NaN
214   NaN
Name: salary, Length: 67, dtype: float64

So, we see that for every value where status is **Not Placed**, the salary is null

The best option in this case is to remove these rows with the status 'Not Placed'